Safrul Lauparenta

safruls opened the pull request:
Kolom komentar ini untuk isi dari file .env dan komentar tambahan lainnya.

.env:
PORT=3000
SECRET=secret123
CLIENT_ID=411524848726-8596p4ossfobk8fd2c3sv0acalu7mpvj.apps.googleusercontent.com

Komentar tambahan:
(nothing to comment at the moment)

Link dari firebase dan heroku:
firebase: https://kanban-uwu.web.app
heroku: https://kanban-uwu.herokuapp.com

safruls committed:
Client:
1. Persiapan file index.html dan main.js
Server:
1. Persiapan file app.js
2. Install package terkait.

safruls committed:
Server:
1. Menambahkan file dokumentasi api-doc.md

safruls committed:
Client:
1. Menambahkan form login dan register di index.html
2. Memperbaiki kanban board di index.html
Server:
1. Melengkapi API documentation dari endpoint route hingga penjelasan detil terkait response status masing-masing endpoint

safruls committed:
Client:
1. Revisi minor di index.html dan main.js untuk persiapan implementasi vue.js
Server:
1. Membuat database 'kanban app'.
2. Menambahkan model User dan Task dengan Foreign Key UserID di tabel Tasks.
3. Menambahkan asosiasi one to many (one User has many Tasks) di model.
4. Menambahkan fungsi helpers untuk enkripsi password saat registrasi user di bcrypt.js.
5. Menambahkan fungsi pemberi dan penelaah access_token di jwt.js.
6. Menambahkan middleware untuk errors handler di errorHandler.js
7. Menambahkan middleware authentication dan authorization di auth.js.
8. Melengkapi end point routes sesuai dengan API Doc dan fitur CRUD untuk tabel Tasks.
9. Menambahkan file userController.js dan taskController.js

ziadyjo commented on server/api-doc.md:
```diff
@@ -0,0 +1,194 @@
+
+&nbsp;
+
+## RESTful Endpoints
```
routing untuk move/ubah categorynya dibuat ya, boleh pake PATCH

ziadyjo commented on server/app.js:
```diff
@@ -0,0 +1,19 @@
+require('dotenv').config()
```
nanti bagian ini dikasih conditional ya ketika mau deploy

ziadyjo replied:
app.js udah mantap

ziadyjo commented on server/config/config.json:
```diff
@@ -0,0 +1,23 @@
+    "host": "127.0.0.1",
+    "dialect": "mysql"
+  },
+  "production": {
```
bagian ini juga jgn lupa diubah kalau udah mau deploy🚀

ziadyjo commented on server/controllers/taskController.js:
```diff
@@ -0,0 +1,152 @@
+                    result.forEach(el => {
+                        if(el.category === 'backlog'){
+                            if(obj[el.category] === undefined){
+                                obj[el.category] = [el]
+                            }
+                            else{
+                                obj[el.category].push(el)
+                            }
+                        }
+                        else if(el.category === 'todo'){
+                            if(obj[el.category] === undefined){
+                                obj[el.category] = [el]
+                            }
+                            else{
+                                obj[el.category].push(el)
+                            }
+                        }
+                        else if(el.category === 'doing'){
+                            if(obj[el.category] === undefined){
+                                obj[el.category] = [el]
+                            }
+                            else{
+                                obj[el.category].push(el)
+                            }
+                        }
+                        else{
+                            if(obj[el.category] === undefined){
+                                obj[el.category] = [el]
+                            }
+                            else{
+                                obj[el.category].push(el)
+                            }
+                        }
+
+                    })
+                    res.status(200).json(obj)
+                }
+                else{
+                    next({name: 'Not Found', message: 'Data not found!'})
+                }
```
ini udah good ya untuk filternya🚀 biar mudah ngefilter di vuenya nanti

ziterz commented on server/controllers/taskController.js:
```diff
@@ -0,0 +1,152 @@
+            })
+    }
+
+    static updateTask(req, res, next){
```
untuk update category pake controller ini kan?

ziadyjo commented on server/controllers/taskController.js:
```diff
@@ -0,0 +1,152 @@
+            })
+    }
+
+    static modifyTaskCategory(req, res, next){
```
ini udh ada ya functionnya, tinggal diupdate aja api doc nya

ziadyjo commented on server/controllers/userController.js:
```diff
@@ -0,0 +1,115 @@
+            email : req.body.email,
+            password: req.body.password
+        }
+        User.findOne({where: {email: obj.email}})
```
keren, di handle sama server juga ya👍🏻

ziadyjo commented on server/controllers/userController.js:
```diff
@@ -0,0 +1,115 @@
+        }
+    }
+
+    static googleLogin(req, res, next){
```
google login udah sesuai sih, nanti pastiin link yang di whitelistnya dikasih ya lewat comment pull request atau di readme

ziadyjo commented on server/middlewares/errorHandler.js:
```diff
@@ -0,0 +1,41 @@
+const errorHandler = (err, req, res, next) => {
```
error handler udah cukup sih, tapi kalau ada yg mau ditambahin lagi boleh yaa

ziadyjo commented on server/middlewares/userAuth.js:
```diff
@@ -0,0 +1,61 @@
+const { User, Task } = require('../models/index.js')
+
+// middleware for user authentication
+const authentication = (req, res, next) => {
```
ga ada masalah ya

ziadyjo commented on server/middlewares/userAuth.js:
```diff
@@ -0,0 +1,61 @@
+}
+
+// middleware for user authorization
+const authorization = (req, res, next) => {
```
ini juga ga ada masalah

ziadyjo commented on server/models/task.js:
```diff
@@ -0,0 +1,56 @@
+        }
+      }
+    },
+    assigned_to: {
```
assign to itu untuk user Id juga ya?

safruls replied:
Bukan kak, itu bisa diisi langsung sama user nanti. Jadi user kan bakal assign suatu task bisa ke diri sendiri atau temannya. Jadi cukup ketik nama aja.

ziadyjo commented on server/models/user.js:
```diff
@@ -0,0 +1,62 @@
+
+  User.beforeCreate((user, options) => {
+    user.password = hashPassword(user.password)
+    user.organization = 'Hacktiv8'
```
good

ziadyjo commented on server/package.json:
```diff
@@ -0,0 +1,21 @@
+  "version": "1.0.0",
+  "description": "",
+  "main": "app.js",
+  "scripts": {
```
dibagian script boleh juga ya kali mau ditambah dev, misal isinya: "NODE_ENV=development nodemon app.js"

ziadyjo commented on server/routes/index.js:
```diff
@@ -0,0 +1,22 @@
+router.get('/tasks', TaskController.getAllTask)
+router.post('/tasks', TaskController.addTask)
+
+router.get('/tasks/:id', authorization, TaskController.getTaskById)
```
untuk get sebetulnya ga perlu authorization ya, biar bisa dilihat aja

ziadyjo reviewed (commented):
Lumayan untuk update hari ini👍🏻

safruls committed:
Client:
1. Menerapkan konsep vue components.
2. Menambahkan fiture CRUD dari sisi client.
3. Menambahkan form edit di EditForm.vue.
4. Menambahkan form add di AddForm.vue.
Server:
1. Memperbaiki static method getAllTask di taskController.js

safruls committed:
Client:
1. Memasukkan folder backup-html ke .gitignore

safruls committed:
final update before release

safruls committed:
finally

safruls committed:
Update API Doc

safruls committed:
Revisi PUT task
